core/deployment/src/main/java/io/quarkus/deployment/dev/IsolatedDevModeMain.java

```diff
@@ -210,6 +224,113 @@ private void startCodeGenWatcher(QuarkusClassLoader classLoader, List<CodeGenDat
         fsWatchUtil.observe(watchers, 500);
     }
 
+    private void startArthasIfNecessary(RunningQuarkusApplication runner) throws IOException {
+        var arthasEnabledOpt = runner.getConfigValue("quarkus.arthas.enabled", Boolean.class);
+        if (arthasEnabledOpt.isEmpty() || !arthasEnabledOpt.get()) {
+            return;
+        }
+
+        var parentDirOpt = runner.getConfigValue("quarkus.arthas.jar-directory", String.class);
+        if (parentDirOpt.isEmpty()) {
+            // should not happen
+            return;
+        }
+        var parentDir = Paths.get(parentDirOpt.get());
+        if (!Files.exists(parentDir)) {
+            Files.createDirectory(parentDir);
+        }
+
+        var arthasHttpPortOpt = runner.getConfigValue("quarkus.arthas.http-port", Integer.class);
+        if (arthasHttpPortOpt.isEmpty()) {
+            // should not happen
+            return;
+        }
+        var arthasHttpPort = arthasHttpPortOpt.get();
+
+        var arthasVersionOpt = runner.getConfigValue("quarkus.arthas.version", String.class);
+        if (arthasVersionOpt.isEmpty()) {
+            // should not happen
+            return;
+        }
+        var arthasVersion = arthasVersionOpt.get();
+        var arthasDir = parentDir.resolve(String.format("arthas-%s", arthasVersion));
+        if (!Files.exists(arthasDir)) {
+            Files.createDirectory(arthasDir);
+        }
+
+        Runnable r;
+        if (!Files.exists(toArthasJar(arthasDir))) {
+            r = () -> startArthas(downloadArthas(arthasDir, arthasVersion), arthasHttpPort);
+        } else {
+            r = () -> startArthas(toArthasJar(arthasDir), arthasHttpPort);
+        }
+        var arthasThread = new Thread(r);
+        arthasThread.setName("Arthas thread");
+        arthasThread.setDaemon(true);
+        arthasThread.start();
+    }
+
+    private Path downloadArthas(Path arthasDir, String version) {
+        var downloadURL = "https://arthas.aliyun.com/download/" + version;
```

Comment:
Can't say I'm excited with us downloading a jar by ourselves from a maybe unstable URL. So first, let's use a Maven Central URL to download this, the artifact is there AFAICS: https://search.maven.org/artifact/com.taobao.arthas/arthas-boot/3.5.5/jar .

Given how it works, I suppose using a dependency is out of the question?

Reply:
That's not enough unfortunately. For it to work, we need the entire zip.

Reply:
I think using an extension and a dependency should be feasible.
You can then locate the JAR from Maven dependency mechanism and launch it.

Reply:
Let me try it again

Reply:
Tried it again and it does not work - the entire zip is needed.

Reply:
https://github.com/alibaba/jbang-catalog/blob/main/jbang-catalog.json#L4

its in maven - its at https://repo1.maven.org/maven2/com/taobao/arthas/arthas-boot/3.5.5/arthas-boot-3.5.5-jar-with-dependencies.jar

and it exists as maven dependency too but you will need to resolve it without transitive dependencies.

Reply:
OK, let me try this.

Reply:
I am not using this.